### DISCUSSION TITLE
A Wildly Inefficient Text Editor

### DISCUSSION THREAD

**Discussion by saikyun** [janet-lang/janet#600] [Show and tell]
I made an inefficient text editor in Janet. Maybe you'll find the code fun to look at. :) Feel free to try it out, mostly tested on MacOS, and sogaiu has run it on Linux. :)

[1 minute presentation](https://www.youtube.com/watch?v=ReUczDAWLfI)

![image](https://user-images.githubusercontent.com/2477927/105085989-d9c49500-5a98-11eb-9aac-40faa59bb285.png)

[Code](https://github.com/Saikyun/text-experiment)

**Comment by pyrmont** [comment]
This is cool! When you say in the video that it's not efficient, is that inevitable or something that could be fixed? :)

**Reply by saikyun** [reply]
Thank you! :)

It's something I want to fix, and I have some ideas about how to solve that. :) But I haven't worked on text editing before, so I need to explore more. Currently I'm thinking of rendering to a texture, and only re-rendering when text changes.

**Comment by uvtc** [comment]
Neat! And I see you used jaylib. I'll have to peek at your code to see what I can learn. Thanks for sharing.

**Comment by saikyun** [comment]
Please do! :) I hope you learn something. Feel free to ask me anything.

Skickat från min iPhone

**Reply by pepe** [reply]
OT, but skickat is my word of the day.

**Reply by saikyun** [reply]
Hahaha, now you know some (more) Swedish. :D

**Reply by pepe** [reply]
Exactly. I have seen some Swedish movies, and I always liked the sound of the language.

**Reply by saikyun** [reply]
Oh really? That’s interesting. What movies do you like? :)

**Reply by pepe** [reply]
One I liked a lot was Fyra nyanser av brunt. The truth is I am not watching movies at all lately—this one I have seen in the club cinema I was frequenting before kids were born.

The language impression I got from Bergman's movies mostly.

**Reply by saikyun** [reply]
Ah, cool. I haven’t watched it, but I like Robert Gustavsson.

Actually haven’t seen anything by Bergman either. I should be ashamed. :’D They just seem so heavy.

**Comment by ColonelPhantom** [comment]
FWIW: Lite XL is a text editor that's quite simple and largely written in Lua. It uses a concept called cached software rendering, which seems to be quite efficient.

**Reply by saikyun** [reply]
I've since made [Freja](https://github.com/saikyun/freja) which is not quite as inefficient, and a lot more powerful. :D But thanks for the tip! Hadn't heard of Lite XL.
